client/src/js/components/bhFindDocument.js

```diff
@@ -0,0 +1,114 @@
+angular.module('bhima.components')
+.component('bhFindDocument', {
+  controller   : FindDocumentComponent,
+  controllerAs : '$ctrl',
+  templateUrl  : 'partials/templates/bhFindDocument.tmpl.html',
+  bindings: {
+    onSearchComplete     : '&',  // bind callback
+    enablePatientDetails : '<',  // bind boolean (true|false) : Enable patient details option
+    enableOptionBar      : '<',  // bind boolean (true|false) : Enable option for add, display list or display thumbnail in a bar
```

Comment:
Can most of these options be bound as attributes `@` to avoid too many `$watch()` statements?  Or must they be dynamic?